### PR TITLE
FE-714 - Add Cypress as a dependency and incorporate necessary support to use locally

### PR DESCRIPTION
**Pull request by nicklemmon**
[FE-714](https://jira.int.messagesystems.com/browse/FE-714)

### What Changed
- Installed Cypress
- Add new npm scripts to run end to end tests
- Add ESLint config unique to cypress to disable undefined variable rule
- Added a few tiny Cypress tests
- Added some documentation to the README explaining how to run Cypress locally
- Added script to generate cypress env config file for authentication purposes
- I have no idea why, but `AccountDropdown.js` was showing an ESLint error...so I fixed it? Unrelated to the other changes...🤷
- Add scripts required to incorporate with CI in the future

### How To Test
- Run `npm install` on this branch
- Take a look at the README and follow the directions...if you can't get Cypress up and running locally then I have failed!

### To Do
- [ ] Incorporate feedback

### CODE REVIEW

File: scripts/generateCypressEnv.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+console.log(`Checking for existing "${file}"...`);
+
+fs.pathExists(file)
+  .then((exists) => {
+    fileExists = exists;
+  })
+  .then(() => {
+    if (fileExists) {
+      console.log(`✅  "${file}" already available, skipping setup and launching Cypress`.green);
```
Comment (kchu1337) [inline scripts/generateCypressEnv.js]: If you just run `npm run test-e2e-generate-env`, it won't start Cypress. Not a big deal but but I waited for like a minute for something to pop up 😆 .

File: cypress/tests/signals/signals.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    cy.contains('Health Score').should('be.visible');
+    cy.contains('Spam Traps').should('be.visible');
+    cy.contains('Engagement Recency').should('be.visible');
+  });
+
+  describe('Health Score', () => {
+    it('renders a health score dashboard after navigating to the health score view', () => {
+      cy.contains('Health Score').click();
+      cy.title().should('include', 'Signals');
+      //cy.title().should('include', 'Health Score'); // NOTE: Commenting out as this is failing - this is a bug!
```
[the remark this answers is not in the document]
Comment (nicklemmon) [reply cypress/tests/signals/signals.js]: It's a bug in our component - the page title should update the Health Score but it doesn't.

File: cypress/tests/account-settings/account-settings.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+/// <reference types="Cypress" />
+
+describe('Account settings', () => {
+  it('is accessible via the user popover in the header', () => {
+    cy.login();
+    cy.contains('appteam@messagesystems.com').click();
```
[the remark this answers is not in the document]
Comment (nicklemmon) [reply cypress/tests/account-settings/account-settings.js]: So I think we will generally want to run our tests with one account for this kind of consistency so I think it's OK?

File: cypress/tests/signals/signals.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (nicklemmon) [reply cypress/tests/signals/signals.js]: I actually just uncommented those - may as well let them fail since this isn't hooked up to CI yet.

**Comment by nicklemmon** [comment]
@kchu1337 Ready for another pass when you have a chance. No rush.

**Pull request merged**